### DISCUSSION TITLE
How does the Timezone work in the HistoricalAPI ?

### DISCUSSION THREAD

**Discussion by AngheloRossi** [open-meteo/open-meteo#801] [Q&A]
Hello everyone,

First of all, I want to thank you all for the great service you are offering with this API.

I'm trying to build my personal WeatherApp and I need some clarity on how the Timezones work with the HistoricAPI. I'm trying to download one year of historical data geolocalized for Milan (IT, Lat: 45.4643, Lon: 9.1895). I'm using daily aggregation and I have set all the parameters to take into consideration the time zone delay, setting the timezone parameter to "Europe/Berlin". However, when I visualized the response, I found out that the output had a static time offset set to "+2h" for all the days in the past year. Shouldn't it be modular, changing from UTC+2 to UTC+1 during the winter days? Am I missing something?

Best regards

**Comment by patrick-zippenfenig** [comment]
Hi,
currently the API will only resolve the current timezone to an offset (1 or 2 hours for CET/CEST) and just it for the entire time range without any DST switch. Your app will have to take care of DST switching if required. It would recommend to use unix timestamps which are always in GMT.

**Reply by malmistry** [reply]
Hi Patrick and team,
Thanks for the amazing effort in making open-meteo such a great data resource. I have generally used Copernicus-CDS or Google Earth Engine, but I find both those quite slow in retrieving historical data such as the ERA5 climate reanalysis. 

I had a related question and instead of opening another thread, I thought of requesting a clarification here. I am extracting the ERA5 daily mean temperature (Tmean) spanning a number of years (1985-2023 for now). Let's say that I am not worried about DST for now and am only interested to extract the daily Tmean accounting for the correct timezone, such as for Milan in the above example. My doubts are:

(i) Is the aggregation to daily from hourly values of ERA5 done during this extraction phase? Or are the retrieved data directly from the daily ERA5 dataset? 

(ii) If latter, I wonder how the timezone parameter (which I am setting to "auto") can represent the correct 24-hour local time averaged temperature. For the daily temperature to be averaged over the location's 24-hour cycle, the hourly data will first need to be shifted to the location-specific timezone (in this case to UTC+01 ignoring the summer time), and then averaged over it (so in this case, over 2300h - 2259h UTC as Milan would be 1 hour ahead). I am not sure if this timezone parameter is simply offsetting the time stamp, but the actual averaged value is always following the native UTC 24-hr cycle.

Thanks and regards,
Malcolm

**Reply by patrick-zippenfenig** [reply]
Hi, ERA5 is download and processed in hourly time steps. Daily data is calculated on demand by shifting hourly data and then aggregating. I am unaware that ERA5 is available in daily resolution directly (only monthly).

Daily values should therefore correctly represent the timezone shift if timezone is set to auto (ignoring DST).
